src/lib/remark-plugin-adjust-link-urls.ts

```diff
@@ -36,6 +37,20 @@ export const preAdjustUrl = ({ currentPath, url }): string => {
 	const urlParts = url.split('/')
 	const currentPathParts = currentPath.split('/')
 
+	// Handle folder-relative URL that is linking upwards
+	if (url.startsWith('../')) {
+		const levelsUpCount = urlParts.filter((part) => part === '..').length
+		return path.join(
+			currentPathParts.slice(0, -(levelsUpCount + 1)).join('/'),
+			urlParts.slice(levelsUpCount).join('/')
```

Comment:
Would be good to annotate with a note or extract into variables here to describe what these are doing.

Reply:
Good idea! Just added two helpers with lots of comments through the steps.

- [`handleDotDotFolderRelativeUrl `](https://github.com/hashicorp/dev-portal/blob/amb.folder-relative-urls/src/lib/remark-plugin-adjust-link-urls.ts#L15-L45)
- [`handleDotSlashFolderRelativeUrl `](https://github.com/hashicorp/dev-portal/blob/amb.folder-relative-urls/src/lib/remark-plugin-adjust-link-urls.ts#L47-L67)